### PR TITLE
do resolve operations concurrently

### PR DESCRIPTION
**Pull request by whyrusleeping**
Retrieve the ipns record and public key concurrently for a bit of speedup on name resolves.

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: namesys/routing.go [hunk 1]
```diff
@@ -132,23 +133,41 @@ func (r *routingResolver) resolveOnce(ctx context.Context, name string) (path.Pa
 	// /ipns/<name>
```
Comment (noffle) [inline namesys/routing.go]: Should the "name should be a multihash" comment be above the block on line 125?

**Comment by noffle** [comment]
LGTM.

Appveyor seems to be confused ("Specify a project or solution file. The directory does not contain a project or solution file."). Known issue?

**Comment by Dignifiedquire** [comment]
appveyor config is not merged yet: https://github.com/ipfs/go-ipfs/pull/2137

File: namesys/routing.go (hunk 1 above)
Comment (whyrusleeping) [inline namesys/routing.go]: ah, probably. I'll go ahead and change that here

File: namesys/routing.go [hunk 2]
```diff
@@ -123,32 +124,50 @@ func (r *routingResolver) resolveOnce(ctx context.Context, name string) (path.Pa
...
+		pubk, err := routing.GetPublicKey(r.routing, ctx, hash)
+		if err != nil {
+			resp <- err
+		}
+		pubkey = pubk
+		resp <- nil
+	}()
+
+	for i := 0; i < 2; i++ {
+		err = <-resp
```
Comment (jbenet) [inline namesys/routing.go]: is this much faster than a `WaitGroup`? (waitgroups are less error prone)

**Comment by jbenet** [comment]
LGTM other than waitgroup comment o/

File: namesys/routing.go (hunk 2 above)
Comment (whyrusleeping) [inline namesys/routing.go]: waitgroups don't allow us to easily return errors. Otherwise its  functionally identical to a waitgroup

Comment (jbenet) [inline namesys/routing.go]: ahhh indeed

**Pull request merged**